src/main/java/com/flowpowered/noise/module/modifier/Range.java

```diff
@@ -0,0 +1,137 @@
+ */
+package com.flowpowered.noise.module.modifier;
+
+import com.flowpowered.noise.exception.NoModuleException;
+import com.flowpowered.noise.module.Module;
+
+/**
+ * A modifier module to map a value from one range to another.
+ * For example: 0.5 (-1 &lt; x &lt; 1) -&gt; 0.75 (0 &lt; x &lt; 1)
+ */
+public class Range extends Module {
+    
+    public static final double DEFAULT_CURRENT_LOWER_BOUND = -1f;
+    public static final double DEFAULT_CURRENT_UPPER_BOUND = 1f;
+    public static final double DEFAULT_NEW_LOWER_BOUND = 0f;
+    public static final double DEFAULT_NEW_UPPER_BOUND = 1f;
+    
+    /* Current lower bound */
+    private double currentLowerBound = DEFAULT_CURRENT_LOWER_BOUND;
+    /* Current upper bound */
+    private double currentUpperBound = DEFAULT_CURRENT_UPPER_BOUND;
+    /* New lower bound */
+    private double newLowerBound = DEFAULT_NEW_LOWER_BOUND;
+    /* New Upper Bound */
+    private double newUpperBound = DEFAULT_NEW_UPPER_BOUND;
+    /* Cache variables */
+    private double scale;
+    private double bias;
+
+    public Range() {
+        super(1);
+    }
+
+    public double getCurrentLowerBound() {
+        return currentLowerBound;
+    }
+
+    // Bounds setters are private to prevent changes to bounds
+    // without recalculating scale/bias
+    private void setCurrentLowerBound(double currentLowerBound) {
+        this.currentLowerBound = currentLowerBound;
+    }
+
+    public double getCurrentUpperBound() {
+        return currentUpperBound;
+    }
+
+    private void setCurrentUpperBound(double currentUpperBound) {
+        this.currentUpperBound = currentUpperBound;
+    }
+
+    public double getNewLowerBound() {
+        return newLowerBound;
+    }
+
+    private void setNewLowerBound(double newLowerBound) {
+        this.newLowerBound = newLowerBound;
+    }
+
+    public double getNewUpperBound() {
+        return newUpperBound;
+    }
+
+    private void setNewUpperBound(double newUpperBound) {
+        this.newUpperBound = newUpperBound;
+    }
+    
+    /*
+     * Calculate the scale and biased to be a applied during Range#getValue(int x, int y, int z)
+     * Should be called when the bounds are modified
+     */
+    private void recalculateScaleBias() {
+    	scale = (getNewUpperBound() - getNewLowerBound()) / 
+        		(getCurrentUpperBound() - getCurrentLowerBound());
+    	bias = getNewLowerBound() - getCurrentLowerBound() * scale;
+    }
+    
+    /**
+     * Configure bounds for range module
+     * @param currentLower current lower bound
+     * @param currentUpper current upper bound
+     * @param newLower new lower bound
+     * @param newUpper new upper bound
+     */
+    public void setBounds(double currentLower, double currentUpper, double newLower,
+    		double newUpper) {
+    	setCurrentLowerBound(currentLower);
+    	setCurrentUpperBound(currentUpper);
+    	setNewLowerBound(newLower);
+    	setNewUpperBound(newUpper);
+    	recalculateScaleBias();
```

Comment:
Try the following situation:
the object is set accordingly:
Range:
    currentLowerBound - 0
    currentUpperBound - 1
    newLowerBound - 0.5
    newUpperBound - 1

and now the programmer want's to reset the bounds to get:
Range:
    currentLowerBound - 1
    currentUpperBound - 2
    newLowerBound - 0.5
    newUpperBound - 1

If the scale/bias recalculation is happening every single setter, the following will happen:
setBounds(currentLower 1, currentUpper 2, newLower 0.5, newLower 1)
    > setCurrentLowerBound(1)
currentUpper is still 1, recalculateScaleBias is called, ArithmaticException is thrown